mteb/evaluation/evaluators/RetrievalEvaluator.py

```diff
@@ -468,51 +466,74 @@ def evaluate(
         qrels: dict[str, dict[str, int]],
         results: dict[str, dict[str, float]],
         k_values: List[int],
-        ignore_identical_ids: bool = True,
+        ignore_identical_ids: bool = False,
     ) -> Tuple[Dict[str, float], dict[str, float], dict[str, float], dict[str, float]]:
         if ignore_identical_ids:
             logger.info(
                 "For evaluation, we ignore identical query and document ids (default), please explicitly set ``ignore_identical_ids=False`` to ignore this."
             )
-            popped = []
+            results_wo_identical_ids = copy.deepcopy(results)
             for qid, rels in results.items():
                 for pid in list(rels):
                     if qid == pid:
-                        results[qid].pop(pid)
-                        popped.append(pid)
+                        results_wo_identical_ids[qid].pop(pid)
 
         ndcg = {}
         _map = {}
         recall = {}
         precision = {}
+        ndcg_no_self = {}
 
         for k in k_values:
+            ndcg_no_self[f"NDCG (no self)@{k}"] = 0.0
             ndcg[f"NDCG@{k}"] = 0.0
             _map[f"MAP@{k}"] = 0.0
             recall[f"Recall@{k}"] = 0.0
             precision[f"P@{k}"] = 0.0
 
         map_string = "map_cut." + ",".join([str(k) for k in k_values])
         ndcg_string = "ndcg_cut." + ",".join([str(k) for k in k_values])
+        ndcg_string_no_self = "ndcg_cut." + ",".join([str(k) for k in k_values])
         recall_string = "recall." + ",".join([str(k) for k in k_values])
         precision_string = "P." + ",".join([str(k) for k in k_values])
         evaluator = pytrec_eval.RelevanceEvaluator(
             qrels, {map_string, ndcg_string, recall_string, precision_string}
         )
         scores = evaluator.evaluate(results)
 
+        # if ignore_identical_ids is True, we evaluate the results without the self-identical ids
+        if ignore_identical_ids:
+            evaluator_no_self = pytrec_eval.RelevanceEvaluator(
+                qrels, {ndcg_string_no_self}
+            )
+            scores_wo_identical_ids = evaluator_no_self.evaluate(
+                results_wo_identical_ids
+            )
+
         for query_id in scores.keys():
             for k in k_values:
                 ndcg[f"NDCG@{k}"] += scores[query_id]["ndcg_cut_" + str(k)]
                 _map[f"MAP@{k}"] += scores[query_id]["map_cut_" + str(k)]
                 recall[f"Recall@{k}"] += scores[query_id]["recall_" + str(k)]
                 precision[f"P@{k}"] += scores[query_id]["P_" + str(k)]
+                if ignore_identical_ids:
+                    ndcg_no_self[f"NDCG (no self)@{k}"] += scores_wo_identical_ids[
```

Comment:
```suggestion
                    ndcg_no_self[f"NDCG(no self)@{k}"] += scores_wo_identical_ids[
```
minor preference (either is fine)